app/browser/reducers/windowsReducer.js

```diff
@@ -4,18 +4,276 @@
+const {zoomLevel} = require('../../common/constants/toolbarUserInterfaceScale')
+
+const isDarwin = process.platform === 'darwin'
+const isWindows = process.platform === 'win32'
+
+// TODO cleanup all this createWindow crap
+function isModal (browserOpts) {
+  // this needs some better checks
+  return browserOpts.scrollbars === false
+}
+
+const navbarHeight = () => {
+  // TODO there has to be a better way to get this or at least add a test
+  return 75
+}
+
+/**
+ * Determine window dimensions (width / height)
+ */
+const setWindowDimensions = (browserOpts, defaults, newWindowState) => {
+  if (newWindowState.ui && newWindowState.ui.size) {
+    browserOpts.width = firstDefinedValue(browserOpts.width, newWindowState.ui.size[0])
+    browserOpts.height = firstDefinedValue(browserOpts.height, newWindowState.ui.size[1])
+  }
+  browserOpts.width = firstDefinedValue(browserOpts.width, browserOpts.innerWidth, defaults.width)
+  // height and innerHeight are the frame webview size
+  browserOpts.height = firstDefinedValue(browserOpts.height, browserOpts.innerHeight)
+  if (typeof browserOpts.height === 'number') {
+    // add navbar height to get total height for BrowserWindow
+    browserOpts.height = browserOpts.height + navbarHeight()
+  } else {
+    // no inner height so check outer height or use default
+    browserOpts.height = firstDefinedValue(browserOpts.outerHeight, defaults.height)
+  }
+  return browserOpts
+}
+
+/**
+ * Determine window position (x / y)
+ */
+const setWindowPosition = (browserOpts, defaults, newWindowState) => {
+  if (browserOpts.positionByMouseCursor) {
+    const screenPos = electron.screen.getCursorScreenPoint()
+    browserOpts.x = screenPos.x
+    browserOpts.y = screenPos.y
+  } else if (newWindowState.ui && newWindowState.ui.position) {
+    // Position comes from window state
+    browserOpts.x = firstDefinedValue(browserOpts.x, newWindowState.ui.position[0])
+    browserOpts.y = firstDefinedValue(browserOpts.y, newWindowState.ui.position[1])
+  } else if (typeof defaults.x === 'number' && typeof defaults.y === 'number') {
+    // Position comes from the default position
+    browserOpts.x = firstDefinedValue(browserOpts.x, defaults.x)
+    browserOpts.y = firstDefinedValue(browserOpts.y, defaults.y)
+  } else {
+    // Default the position
+    browserOpts.x = firstDefinedValue(browserOpts.x, browserOpts.left, browserOpts.screenX)
+    browserOpts.y = firstDefinedValue(browserOpts.y, browserOpts.top, browserOpts.screenY)
+  }
+  return browserOpts
+}
+
+function windowDefaults (state) {
+  setDefaultWindowSize(state)
+
+  return {
+    show: false,
+    width: state.getIn(['defaultWindowParams', 'width']) || state.get('defaultWindowWidth'),
+    height: state.getIn(['defaultWindowParams', 'height']) || state.get('defaultWindowHeight'),
+    x: state.getIn(['defaultWindowParams', 'x']) || undefined,
+    y: state.getIn(['defaultWindowParams', 'y']) || undefined,
```

Comment:
should we use `state.getIn(['defaultWindowParams', 'y'], undefined)` here? and for all examples above?